src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/FeatureTagHelperTypeResolver.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.AspNetCore.Razor.Runtime.TagHelpers;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    /// <summary>
+    /// Resolves tag helper types from the <see cref="ApplicationPartManager.ApplicationParts"/>
+    /// of the application.
+    /// </summary>
+    public class FeatureTagHelperTypeResolver : ITagHelperTypeResolver
+    {
+        private readonly ApplicationPartManager _manager;
+
+        /// <summary>
+        /// Initializes a new <see cref="FeatureTagHelperTypeResolver"/> instance.
+        /// </summary>
+        /// <param name="manager">The <see cref="ApplicationPartManager"/> of the application.</param>
+        public FeatureTagHelperTypeResolver(ApplicationPartManager manager)
+        {
+            if (manager == null)
+            {
+                throw new ArgumentNullException(nameof(manager));
+            }
+
+            _manager = manager;
+
+        }
+
+        /// <inheritdoc />
+        public IEnumerable<Type> Resolve(string assemblyName, SourceLocation documentLocation, ErrorSink errorSink)
```

Comment:
It's a good suggestion, I think it's possible if we also override IsType, make it return true and seal it. We want the source of truth to be the TagHelperFeatureProvider. I'll do it unless someone else has an objection to it.